### ISSUE TITLE
'this' is undefined on the server side

### ISSUE THREAD

**Issue by SoonDead** [reactjs/React.NET#176]
Upon upgrading to 2.0 the server side rendering broke:

To render numbers and dates in an i18n compatible way I'm using numeral.js and moment.js. Both of them are written in a way that uses `this` as the global object, like 

    (function () {
        // here, `this` means `window` in the browser, or `global` on the server
        // add `numeral` as a global object via a string identifier,
        // for Closure Compiler 'advanced' mode
        this['numeral'] = numeral;
    }).call(this);

Or

    (function (global, factory) {
        typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
        typeof define === 'function' && define.amd ? define(factory) :
        global.moment = factory()
    }(this, function () {
        var _moment;
        ......
        return _moment;
    });

They are both valid in a browser or a node.js environment, but the following fails:

ReactConfig.cs:

    ReactSiteConfiguration.Configuration
        .AddScript("~/Scripts/numeral.js")
        .AddScript("~/Scripts/my-component.js");

Index.cshtml:

    @Html.React("MyComponent", new { prop1 = "prop" })

Since it cannot set the property 'numeral' of undefined.

This pretty much blocks me from upgrading to React.Core 2.0 and therefore React 0.14.

**Comment by Daniel15** [comment]
Thanks for the report, I'll take a look this weekend. This may be due to how Babel handles `this` compared to JSXTransformer.

**Comment by Daniel15** [comment]
Looks like I need to blacklist the "strict" transformer. From [the Babel docs](https://babeljs.io/docs/faq/#why-is-this-being-remapped-to-undefined-):

> ### Why is this being remapped to undefined?

> Babel assumes that all input code is an ES2015 module. ES2015 modules are implicitly strict mode so this means that top-level this is not window in the browser nor is it exports in node.

> If you don't want this behaviour then you have the option of disabling the strict transformer:

> ```
$ babel --blacklist strict script.js
```
> ```js
require("babel").transform("code", { blacklist: ["strict"] });
```

You could try hacking this into [the ReactJS.NET Babel config](https://github.com/reactjs/React.NET/blob/master/src/React.Core/BabelConfig.cs#L63): 

```csharp
{"blacklist", new List<string> { "strict" } }
```

If you do try that out, let me know if it works :)

**Comment by Daniel15** [comment]
Fixed in 2.0.1, thanks for reporting!

**Comment by SoonDead** [comment]
Works perfectly, thanks.
